solr/bin/solr

```diff
@@ -1054,31 +1054,13 @@ fi
 
 # Establish default GC logging opts if no env var set (otherwise init to sensible default)
 if [ -z "${GC_LOG_OPTS}" ]; then
-  if [[ "$JAVA_VER_NUM" -lt "9" ]] ; then
-    GC_LOG_OPTS=('-verbose:gc' '-XX:+PrintHeapAtGC' '-XX:+PrintGCDetails' \
-                 '-XX:+PrintGCDateStamps' '-XX:+PrintGCTimeStamps' '-XX:+PrintTenuringDistribution' \
-                 '-XX:+PrintGCApplicationStoppedTime')
-  else
-    GC_LOG_OPTS=('-Xlog:gc*')
-  fi
-else
-  # TODO: Should probably not overload GC_LOG_OPTS as both string and array, but leaving it be for now
-  # shellcheck disable=SC2128
-  GC_LOG_OPTS=($GC_LOG_OPTS)
+  GC_LOG_OPTS=('-Xlog:gc*')
 fi
 
 # if verbose gc logging enabled, setup the location of the log file and rotation
 if [ "${#GC_LOG_OPTS[@]}" -gt 0 ]; then
-  if [[ "$JAVA_VER_NUM" -lt "9" ]] || [ "$JAVA_VENDOR" == "OpenJ9" ]; then
-    gc_log_flag="-Xloggc"
-    if [ "$JAVA_VENDOR" == "OpenJ9" ]; then
-      gc_log_flag="-Xverbosegclog"
-    fi
-    if [ -z ${JAVA8_GC_LOG_FILE_OPTS+x} ]; then
```

Comment:
what does line 1077 mean?  Is it possible that it saying that if this variable "JAVA8_GC_LOG_FILE_OPTS" isn't set then we do the the GC_LOG_OPTS?

Reply:
From my understanding that is right. The reason I removed this check is because we do not support Java 8, and as the environment variable indicates, this is only relevant for users running SOlr with Java 8 and setting this value.

I couldn't find any references in the documentation, so since it is not documented, I assume that this value is always not set for any Solr version 10+ that only supports Java 21+, and the condition is therefore always `true`.

If a condition is here relevant, we should definitely use a suitable name other than `JAVA8_*` and reintroduce the check.

Reply:
That makes sense.   I do sort of wish we had a bit more ways of testing these settings in our `.bats` tests, especially as with Java major versions coming out every six months, we may find we are using more of these configuration options.   Your cleanup makes sense to me!

Reply:
Yeah, I think this requires a combination of benchmarking and testing to also tweak the configurations in the future. But that's not (yet) my expertise.